File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> outerArr = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<>();
        if(root == null)return outerArr;
        q.offer(root);
        while(!q.isEmpty()){
            int size = q.size();
            List<Integer>innerArr = new ArrayList<>();
            for(int i=0;i<size; i++){
                TreeNode node = q.poll();
                innerArr.add(node.val);
                if(node.left != null){
                    q.offer(node.left);
                }
                 if(node.right != null){
                    q.offer(node.right);
                }

            }
            outerArr.add(innerArr);
        }
        return outerArr;
    }
}

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        outerArr = []
        if not root:
            return root
        q = deque()
        q.append(root)

        while q:
            innerArr = []
            for i in range(len(q)):
                n = q.popleft()
                innerArr.append(n.val)
                if n.left:
                    q.append(n.left)
                if n.right:
                    q.append(n.right)
            outerArr.append(innerArr)
        return outerArr